src/Bicep.Core/TypeSystem/ResourceParentType.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Linq;
+using Bicep.Core.Extensions;
+using Bicep.Core.Resources;
+
+namespace Bicep.Core.TypeSystem
+{
+    public class ResourceParentType : TypeSymbol
```

Comment:
@anthony-c-martin I ended up adding a new type for the parent property and doing validation in `TypeValidator`. I tried to modify `ResourceMetadataCache` to let it emit diagnostics, but I didn't find a good way to return them to the caller other than adding a `IDiagnostic` property to `ResourceMetadata` (which feels a bit hacky since it's only added for an edge case).